src/HoC.js

```diff
@@ -0,0 +1,83 @@
+import React from 'react';
+import {ensure} from './util';
+import {update, unmount} from './Mixin';
```

Comment:
Yeah that makes sense to refactor it so they both share the same code from a common base instead of HoC depending on Mixin. I'm fine deferring that to after this PR if that makes things simpler though.